src/Traits/JsonSerializer.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Softonic\GraphQL\Traits;
+
+trait JsonSerializer
+{
+    public function jsonSerialize(): array
+    {
+        $item = [];
+        if ($this->hasChanged()) {
```

Comment:
To avois some nestig you can use guard clause 

https://engineering.helpscout.com/reducing-complexity-with-guard-clauses-in-php-and-javascript-74600fd865c7